packages/gauges/src/tests/config.test.ts

```diff
@@ -0,0 +1,45 @@
+import { chainNames } from '@pancakeswap/chains'
+import { FACTORY_ADDRESS_MAP, Token, computePairAddress } from '@pancakeswap/sdk'
+import { DEPLOYER_ADDRESSES, computePoolAddress } from '@pancakeswap/v3-sdk'
+import groupBy from 'lodash/groupBy'
+import { describe, expect, it } from 'vitest'
+import { CONFIG_PROD } from '../constants/config/prod'
+import { GaugeType } from '../types'
+
+describe('Gauges Config', async () => {
+  const chainName = chainNames[1]
+
+  const gidGroups = groupBy(CONFIG_PROD, 'gid')
+  Object.entries(gidGroups).forEach(([gid, gauge]) => {
+    it(`${chainName} gauges with gid #${gid} should be unique`, () => {
+      expect(gauge.length).toBe(1)
+    })
+  })
+
+  CONFIG_PROD.forEach((gauge) => {
+    it(`${chainName} gid #${gauge.gid} tokens chainId-lpAddress-feeTier should be matched`, () => {
+      if (gauge.type === GaugeType.V3) {
+        const tokenA = new Token(gauge.chainId, gauge.token0Address, 18, '')
+        const tokenB = new Token(gauge.chainId, gauge.token1Address, 18, '')
```

Comment:
Token decimals will not always be `18`

Reply:
just for instantiation `Token`, it will not affect, Token is just used for sort, decimals actually not used